web/client/codechecker_client/cmd/store.py

```diff
@@ -576,6 +577,28 @@ def assemble_zip(inputs, zip_file, client, checker_labels: CheckerLabels):
     # Print statistics what will be stored to the server.
     stats.write()
 
+    # Fail store early if too many reports.
+    p = prod_client.getCurrentProduct()
+    if stats.num_of_reports > p.reportLimit:
+        over_limit = stats.num_of_reports - p.reportLimit
+        LOG.error(f"""Report Limit Exceeded
+
+This report folder cannot be stored because the number of reports in the
+result folder is too high. This measeure is in place to protect the server
+from overloading.
+
+Please review the 'Checker Statistics' section of this output to gain a better
+understanding of which checkers have generated an excessive number of reports.
+
+Disable checkers that have generated an excessive number of reports and then
+rerun the analysis to be able to store the results on the server.
+
+* Configured report limit for this product: {p.reportLimit}
+* Number of reports in the result folder: {stats.num_of_reports}
+* The report folder exceeds the limit by {over_limit} reports
```

Comment:
There are a few things I dislike about this message:
* I dislike the emphasis on the why we implemented this. Sure, its important not to overload the server, but the server could actually take this load if we really wanted it to. What is more true (at the very least, it surely sounds a lot better) is that the fact that we see half a million reports is a **major** sign that something is off. Either a checker is faulty or should not have been enabled in the first place. The reason why we disallow this by default is that no sane person will look at that many reports one-by-one.
* How can I increase the limit if I believe that it should be? Is this information missing intentionally (I can understand if it is)?
* Can we plug #4018 in this message as well?
* Its not true that this cannot be stored, or thats not the complete story -- it was rejected by the server.

Reply:
* Sure!
* Thanks!
* My reasoning is that `--summary` should be orders of magnitude faster. It seems appropriate to advertise it since both of these features land in the same release, so it'd be reasonable to assume that even powerusers are unaware of it.
* I think we should open the message with "The server rejected this store operation" instead of simply "This can't be stored"

Reply:
@dkrupp  suggested the same that we should not mention the server protection angle. I will remove it. Also added the reasoning behind noisy checkers.